app/controllers/calendar_controller.rb

```diff
@@ -0,0 +1,27 @@
+require 'net/http'
+require 'uri'
+require 'json'
+require 'petfinder'
+
+class CalendarController < ApplicationController
+  def dogs
+    allDogs = []
+    Rails.cache.fetch('cachedDogs', expires_in: 1.hours) do
+      currentPage = 1
+      petfinder = Petfinder::Client.new(ENV.fetch("PETFINDER_API_KEY"), ENV.fetch("PETFINDER_SECRET"))
```

Comment:
This works, but if you put your secrets in `secrets.yml` then you can also call (more conventionally) `Rails.application.secrets` and reference the relevant key. Otherwise if you want to stick with referencing `ENV` you can skip updating `secrets.yml`. Both approaches are valid, but I generally put all my secrets in `secrets.yml` and have that reference `.env`. This app can also benefit from a `.env.template` file that lists all the keys.